jvm/src/main/scala/uk/gov/homeoffice/drt/db/dao/StatusDailyDao.scala

```diff
@@ -0,0 +1,42 @@
+package uk.gov.homeoffice.drt.db.dao
+
+import slick.dbio.Effect
+import uk.gov.homeoffice.drt.db.Db.slickProfile.api._
+import uk.gov.homeoffice.drt.db.serialisers.StatusDailySerialiser
+import uk.gov.homeoffice.drt.db.{StatusDaily, StatusDailyTable}
+import uk.gov.homeoffice.drt.ports.PortCode
+import uk.gov.homeoffice.drt.ports.Terminals.Terminal
+import uk.gov.homeoffice.drt.time.LocalDate
+
+import java.sql.Timestamp
+import scala.concurrent.ExecutionContext
+
+
+object StatusDailyDao {
+  val table: TableQuery[StatusDailyTable] = TableQuery[StatusDailyTable]
+
+  def insertOrUpdate(portCode: PortCode): StatusDaily => DBIOAction[Int, NoStream, Effect.Write] =
+    row =>
+      if (row.portCode == portCode) table.insertOrUpdate(StatusDailySerialiser.toRow(row))
+      else DBIO.successful(0)
+
+  def setUpdatedAt(portCode: PortCode): (StatusDailyTable => Rep[Option[Timestamp]]) => (Terminal, LocalDate, Long) => DBIOAction[Int, NoStream, Effect.Write] =
+    columnToSet => (terminal, date, updatedAt) =>
+      table
+        .filter(_.port === portCode.iata)
```

Comment:
could this be more efficient ? 

```columnToSet => (terminal, date, updatedAt) =>
  table
    .filter(row => 
      row.port === portCode.iata &&
      row.terminal === terminal.toString &&
      row.dateLocal === date.toISOString
    )
    .map(columnToSet)
    .update(Some(new Timestamp(updatedAt)))
```

Reply:
Yes good call. And since the code is repeated I'll also move it to a function